src/components/ReportActionItem/MoneyRequestView.js

```diff
@@ -115,12 +114,17 @@ function MoneyRequestView({report, parentReport, parentReportActions, policyCate
     // Flags for allowing or disallowing editing a money request
     const isSettled = ReportUtils.isSettled(moneyRequestReport.reportID);
     const isCancelled = moneyRequestReport && moneyRequestReport.isCancelledIOU;
+
+    // Used for non-restricted fields such as: description, category, tag, billable, etc.
     const canEdit = ReportUtils.canEditMoneyRequest(parentReportAction);
-    const canEditAmount = canEdit && !isSettled && !isCardTransaction;
-    const canEditReceipt = ReportUtils.canEditFieldOfMoneyRequest(parentReportAction, moneyRequestReport.reportID, CONST.EDIT_REQUEST_FIELD.RECEIPT);
+    const canEditAmount = ReportUtils.canEditFieldOfMoneyRequest(parentReportAction, CONST.EDIT_REQUEST_FIELD.AMOUNT);
+    const canEditMerchant = ReportUtils.canEditFieldOfMoneyRequest(parentReportAction, CONST.EDIT_REQUEST_FIELD.MERCHANT);
+    const canEditDate = ReportUtils.canEditFieldOfMoneyRequest(parentReportAction, CONST.EDIT_REQUEST_FIELD.DATE);
+    const canEditReceipt = ReportUtils.canEditFieldOfMoneyRequest(parentReportAction, CONST.EDIT_REQUEST_FIELD.RECEIPT);
+    const canEditDistance = ReportUtils.canEditFieldOfMoneyRequest(parentReportAction, CONST.EDIT_REQUEST_FIELD.DISTANCE);
 
     // A flag for verifying that the current report is a sub-report of a workspace chat
-    const isPolicyExpenseChat = useMemo(() => ReportUtils.isPolicyExpenseChat(ReportUtils.getRootParentReport(report)), [report]);
+    const isPolicyExpenseChat = policy.type && policy.type !== CONST.POLICY.TYPE.PERSONAL;
```

Comment:
I don't understand how confirming that the current report is not on a personal policy ensures that it is the sub-report of a workspace chat.

Reply:
Since we're currently dealing with a money request report, they all have a policy tied to them. IOUs have a personal policy, and expenses have a policy that is either Free, Collect, or Control. So if the policy is not personal, then we're 100% the report belongs to workspace chat!

Reply:
I'll add a comment

Reply:
Cool cool. Thanks for clarifying.

Reply:
Sorry, I suggested this change earlier similar to [this here] in `EditRequestPage`. But policy in the `EditRequestPage` is found from `getPolicy` function and policy is assured to be not undefined. But here in this `MoneyRequestView` it is passed as a prop and there is no default prop for this and [here](https://github.com/Expensify/App/blob/273fa79f28d083ac69603dd633af6e030cc30ed2/src/pages/home/report/ReportActionItem.js#L576-L579) in `ReportActionItem` policy is not passed and this `policy.type` check without `policy` check crashes the app.

Either add a default prop for policy as {} or check policy before checking for policy.type.

Reply:
Please fix this ^ @youssef-lr

Reply:
Ahh right. Done!